src/Modules/DatePickerInput.react.js

```diff
@@ -1,302 +1,286 @@
+                                color={isCalendarOpen ? 'highlight' : disabled ? 'primary' : null}
+                                compact
+                                disable={disabled}
+                                onClick={this._onIconClick}
+                                type="calendar"
+                            />
+                        )}
+                        keepCharPositions
+                        label={label}
+                        mask={[ /\d/, /\d/, '/', /\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/ ]}
+                        onBlur={this._onInputBlur}
+                        onChange={this._onInputChange}
+                        onFocus={this._onInputFocus}
+                        onKeyDown={this._onInputKeyDown}
+                        placeholder="mm/dd/yyyy"
+                        ref={ref => this._datePickerInput = ref}
+                        required={required}
+                        tabIndex={tabIndex}
+                        type="text"
+                        value={inputValue}
+                    />
+
+                    {isCalendarOpen ? (
+                        <CalendarOnClickOutisde
+                            date={date}
+                            dateFrom={dateFrom}
+                            dateTo={dateTo}
+                            events={events}
+                            excludeDates={excludeDates}
+                            filterDates={filterDates}
+                            includeDates={includeDates}
+                            locale={locale}
+                            maxDate={this._getMaxDate()}
+                            minDate={this._getMinDate()}
+                            mode="input"
+                            onChange={this._onCalendarChange}
+                            onClose={this._onCalendarClickOutside}
+                            onMonthChange={this._onMonthChange}
+                            rangeFrom={rangeFrom}
+                            rangeTo={rangeTo}
                         />
-                    </Grid.Column>
-                </Grid>
-            );
-        }
+                    ) : null}
+                </TetherComponent>
+            </div>
+        );
     }
 
     componentDidMount() {
-        if (this.state.listensForErrors) {
-            ParsleyFormStore.addChangeListener(this._onErrorRef);
+        const { onChange, rangeFrom, rangeTo } = this.props;
+
+        if ((rangeFrom || rangeTo) && _.isUndefined(onChange)) {
+            console.error('The onChange prop is required when using the rangeFrom or rangeTo props');
         }
     }
 
-    componentWillUnmount() {
-        if (this.state.listensForErrors) {
-            ParsleyFormStore.removeChangeListener(this._onErrorRef);
+    _getAllowedDateFormats(specifiedFormat) {
+        const formats = DateUtils.getAllowedDateFormats();
+
+        if (_.indexOf(formats, specifiedFormat) < 0) {
+            formats.unshift(specifiedFormat);
```

Comment:
Hmm, should we consider returning a _mutated copy_ of the original array, as opposed to _actually mutating_ the original?  You probably have to then declare it above using `let` rather than `const` and then rewrite this line as `formats = [ specifiedFormat, ...formats ];`

Reply:
Changed.

Reply:
This seems to be an odd mix of old and new...?
I think it should be as follows:
```JavaScript
_getAllowedDateFormats(specifiedFormat) {
    let formats = DateUtils.getAllowedDateFormats();

    if (_.indexOf(formats, specifiedFormat) < 0) {
        formats = [ specifiedFormat, ...formats ]; // This creates a new array with `specifiedFormat` at the beginning
    }

    return formats;
}
```

Reply:
Side Note on `DateUtils` in general:
At some point, I like to sync up and consolidate React CM UI's `DateUtils` and Healthy Church's `DateUtils` and remove duplicated code.  In other words, I think the Healthy Church back office client application should mostly be using React CM UI's `DateUtils` and if it needs stuff not provided in there (e.g. the logic about checking the preferred locale and time zone from a User's Preferences, which is specific to HC) then it should define its own "extension" `DateUtils` that add such functionality.   Thoughts...?

Reply:
Thank you @groberts314 for your help on this one. Part tired, part not sure what was going on, on my part.
I agree we should sync up the DateUtils. I can do that with another commit soon.

Reply:
Umm, maybe not necessarily as a part of this PR ... that's probably a separate effort I'd think.  More like technical debt.  My thought was to be more efficient and not have duplicate code in the bundles of the HC Back Office app, but it's not necessarily on fire right now this minute.  Let's identify it as technical debt, and plan for it, and do it at the right time.  Up to you.

The only thing in my two previous comments in this thread that's actually actionable right now as part of this PR is fixing the function (which I shouldn't have bothered you about in the first place :-) by either reverting it to the way you had it first, or adapting my suggested version ... up to you.   
 Otherwise, I think the scope of this PR and the forthcoming React CM UI release that contains it is large enough ... we can solve future problems in the future.  Thanks.